### DISCUSSION TITLE
Paypal importer

### DISCUSSION THREAD

**Discussion by Pyrrha** [firefly-iii/firefly-iii#4323] [Firefly III]
Hello,

**Description**

Adding PayPal integration through their API.

**Solution**

A handler to another type of account, that can be feed by another importer.

**Additional context**

I saw a PayPal developer portail 🤷‍♂️

**Bonus points**

I'll be happy 😃

**Comment by JC5** [comment]
Feel free to build one 😁. With the API in place, I'm not focussing on new importers right now.

**Comment by Pyrrha** [comment]
Nice I didn't see the API!

I'll keep you informed here for deployment progress. I'll need it for personnal use and an association I'm working it is also interested by this type of import.

**Reply by Syphdias** [reply]
Any progress on that?

I stumbled upon https://github.com/firefly-iii/firefly-iii/issues/1630 but it's just not as cut and dried in my case. An importer that gets all transactions in FF, deduplicates and converts withdraws/deposits to transactions would be great!

**Comment by robvankeilegom** [comment]
For those interested: I created an importer that imports data from the PayPal API and pushes it to the Firefly API.

https://github.com/robvankeilegom/firefly-III-paypal-importer

**Reply by JC5** [reply]
Nice! Shall I list it in the docs?

**Reply by robvankeilegom** [reply]
Thanks, I created https://github.com/firefly-iii/docs/pull/116

**Reply by martinspaniol** [reply]
According to [#10](https://github.com/robvankeilegom/firefly-III-paypal-importer/issues/10) a paypal merchant account is needed to create API credentials. Maybe there should be a hint in the documentation for this.

**Comment by terrencetingleberry** [comment]
@JC5 Any new thoughts on implementing a PayPal importer directly into data-importer?
